packages/design-system/src/components/EmptyState/primitive/EmptyStatePrimitive.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { forwardRef, HTMLAttributes, ReactElement, Ref } from 'react';
+import { useTranslation } from 'react-i18next';
+import { StackVertical } from '../../Stack';
+
+import Link from '../../Link';
+import { ButtonPrimary } from '../../Button';
+import { ButtonPrimaryAsLink } from '../../ButtonAsLink';
+import { ButtonPrimaryPropsType } from '../../Button/variations/ButtonPrimary';
+import { ButtonPrimaryAsLinkPropsType } from '../../ButtonAsLink/variations/ButtonPrimaryAsLink';
+
+import styles from './EmptyStatePrimitive.module.scss';
+
+type CallbackTypes =
+	| (ButtonPrimaryPropsType & { callbackType: 'button' })
+	| (ButtonPrimaryAsLinkPropsType & { callbackType: 'link' });
+
+export type EmptyStatePrimitiveProps = Omit<
+	HTMLAttributes<HTMLElement>,
+	'className' | 'style' | 'title'
+> & {
+	title: string;
+	description?: string;
+	docLinkURL?: string;
+	callback?: CallbackTypes;
+	illustration?: ReactElement;
+};
+
+function buildCallBack(callback: CallbackTypes) {
+	if (callback.callbackType === 'button') {
```

Comment:
type was not enough?
```suggestion
	if (callback.type === 'button') {
```

Reply:
Oh ok I understand because of the conflict with type="button" if it's a button*

Reply:
Yep :)